### ISSUE TITLE
[admin][UBP] Grant free credits to users in admin dashboard

### ISSUE THREAD

**Issue by jldec** [gitpod-io/gitpod#14591]
Admins should be able to grant additional credits to users E.g. to unblock their usage if they had out-of-control prebuilds before switching to UBP.

Example [here](https://gitpod.slack.com/archives/C027AUU7BC3/p1668112223326439) (internal)

**Comment by jldec** [comment]
A reasonable option would be to reset the clock on their free tier so that they start with a fresh 500 credits.

**Comment by svenefftinge** [comment]
We should allow admins to do these things:
- [ ] Add usage entries to adjust the current credit balance in both directions.
  - they need a special usage kind as well as a mandatory description and userId of the admin.
- [ ] Adjust spending limit of cost-center
- [ ] See usage

@jldec What is the order in which you want to see this in production?
@gtsiolis We need design input on how to switch between different admin areas per team detail (currently we have one page where we list the members).

**Comment by jldec** [comment]
@svenefftinge priorities from product side are:

1. unblock users who hit their limit due to unforeseen reasons like runaway prebuilds - this is probably best done by adjusting the credit balance, but could also be done with a **temporary** usage limit bump.
2. grant a **permanent** increase/decrease in free tier usage limit e.g. for open source maintainers.
3. see current total usage, credit limit, next reset date
4. see detailed usage (workspace sessions)

**Comment by svenefftinge** [comment]
Unassigning myself, because we need input from @gitpod-io/product-design

**Comment by svenefftinge** [comment]
Also see [#14367](https://github.com/gitpod-io/gitpod/issues/14367)

**Comment by jldec** [comment]
From discussion with @gtsiolis 

- action uses standard hypertext link to Grant extra credits
- Trigger modal form for both, with action/Cancel buttons
- Requires a "Reason" to be entered by the admin 
- Free tier, individual plans only
- One time grant only - entered into ledger
- Enter amount (typically 100 or 200)
- Effect will be to reduce "Credits Used" (number should decrease after)
- nice to have: don't allow granting more than current usage (avoid negative result)
- future enhancement: show extra credit grant in the billing or usage view

**Comment by gtsiolis** [comment]
Adding below some visuals to help with the implementation. Cc @svenefftinge @jldec 

| User (Data) | User (Modal) |
|-|-|
| <img width="1440" alt="User" src="https://user-images.githubusercontent.com/120486/201985685-18fb07f2-1636-483e-819c-1087c6e7a8f3.png"> | <img width="1440" alt="User-1" src="https://user-images.githubusercontent.com/120486/201985771-d282bac8-ed96-4124-9de5-1dc0a1559f83.png"> |

FWIW:
- This could reuse a similar pattern we recently used in https://github.com/gitpod-io/gitpod/pull/12805#discussion_r1006734427 for surfacing when a user is blocked from upgrading to UBP to surface the history of credits changes. Cc @geropl 
- The same pattern could be used for adding relevant notes when blocking a user, see https://github.com/gitpod-io/gitpod/issues/10948. Cc @jimmybrancaccio 

| Link Button | Dropdown (Single Entry) | Dropdown (Multiple Entries) 
|-|-|-|
| <img width="1440" alt="User-2" src="https://user-images.githubusercontent.com/120486/201986087-38dbba13-3671-417d-826b-a021506ddc6d.png"> | <img width="1440" alt="User-3" src="https://user-images.githubusercontent.com/120486/201986207-580240e5-8a59-47ec-83e7-f6ea1ede1672.png"> | <img width="1440" alt="User-4" src="https://user-images.githubusercontent.com/120486/201986215-f0077758-fbaf-4a69-a8bf-a97a9e5b35b8.png"> |

See [design specs](https://www.figma.com/file/kjQ4I8TgFCGDsWuDQBoxMU/Deprecate-the-creation-of-projects-under-individual-accounts-%2313414-(Copy)?node-id=2102%3A829&t=c4hb8dhPLQGWwTGd-1).

**Comment by jimmybrancaccio** [comment]
This looks super cool @gtsiolis! 🧡 The only suggestion I have is to show the full "audit log" entry instead of an excerpt. Perhaps make the log element or 'container' the same width as the 2 elements above it (balance used and cycle resets).

**Comment by gtsiolis** [comment]
Thanks @jimmybrancaccio!

Indeed, surfacing all the information sounds better. I deliberately added this as this could out of the scope of this iteration.

Alternatively, using a tooltip for the full note entry could be nice, as long as a user can hover and click any links inside the note. In any case, it could be interesting to also enforce a max character limit so that notes don't span much. This way we could surface all the note. Additionally, using a larger width for the dropdown could help.

Full log entries could be later on surfaced in https://github.com/gitpod-io/gitpod/issues/11469.

**Comment by jimmybrancaccio** [comment]
@gtsiolis Tooltip would be great. Not having the full entry could make this somewhat pointless especially if the more useful information is at the end of the log entry which ends up getting truncated. If we do initially launch this with the log entry truncated we should make sure that anyone who leaves notes knows to put the important content first so it doesn't get truncated.

**Comment by gtsiolis** [comment]
@jimmybrancaccio Thinking about this again, I'd a) avoid the tooltip, b) increase dropdown width, c) use fixed scrollable dropdown height as we do with workspace changes so that all information can be accessed, and let users be conscious about this field if needed.

**Comment by svenefftinge** [comment]
The credit notes would end up as a usage entry. So I propose we just show the whole usage data in a list.
WDYT of adding a tab on top of the workspaces list (members for teams) and add a tab "usage"?

**Comment by jldec** [comment]
@svenefftinge I agree with having a separate usage detail list view, but I would split that out and try to finish the credit adjust first.

I have created 2 issues for followup.  
-  #14760
-  #14762 

cc: @gtsiolis

**Comment by svenefftinge** [comment]
Sure, I meant to say that I will not add the dropdown part that @gtsiolis and @jimmybrancaccio were talking about.

**Comment by gtsiolis** [comment]
Sounds good, @svenefftinge!

> The credit notes would end up as a usage entry.

Do we need any designs for this yet?

**Comment by svenefftinge** [comment]
> Do we need any designs for this yet? We may also need to inform** admins that their submitted information on the note will be visible to team owners, or any other with access to team usage, right?

Yes & Yes! 😅

**Comment by gtsiolis** [comment]
On it. ✋

**Comment by gtsiolis** [comment]
I'm adding below another iteration of the designs taking into account the comment above in https://github.com/gitpod-io/gitpod/issues/14591#issuecomment-1318527904.

Cc @svenefftinge @jldec 

| Add Credits note (Diff:  Rephrased 🅰️ alert and 🅱️ help text) |
|-|
| <img width="1440" alt="User" src="https://user-images.githubusercontent.com/120486/202464359-cd8a0e95-61e5-4f18-b05d-281a92f9d74e.png"> |

For surfacing this information, I tried adding the credit notes as a usage entry and include the whole information or a generic text.

| Usage entry (full text) | Usage entry (generic text) |
|-|-|
| <img width="995" alt="UsageNote" src="https://user-images.githubusercontent.com/120486/202465723-e2314123-179b-4d53-a81c-2ac63c4453ad.png"> | <img width="995" alt="UsageNote-1" src="https://user-images.githubusercontent.com/120486/202465223-839d0d48-71fb-42c8-8fea-982a452f18e3.png"> |

Alternatively, we could use the billing page as the single source of truth so that it's easier for users to understand why they may be seeing a difference in the balance used, etc.

| Billing (link button) | Billing (dropdown) |
|-|-|
| <img width="696" alt="Team Plans" src="https://user-images.githubusercontent.com/120486/202465904-0845d793-703b-4f94-9bb8-4e4ae90781a7.png"> | <img width="696" alt="Team Plans-1" src="https://user-images.githubusercontent.com/120486/202465914-b96df07a-b159-430c-88c5-463e51083efc.png"> |

**Comment by jimmybrancaccio** [comment]
Looks great @gtsiolis! How would one see the credit notes for all time? I see it looks like the dropdown would _only_ show the notes from the current billing cycle?
